xarray/core/dataset.py

```diff
@@ -2832,6 +2832,7 @@ def interp(
         method: str = "linear",
         assume_sorted: bool = False,
         kwargs: Mapping[str, Any] = None,
+        method_for_non_numerics: str = "nearest",
```

Comment:
I like this new kwarg. Shall we do `method_non_numeric` for slightly less typing?

Reply:
Done.